tests/test_mapping.py

```diff
@@ -20,13 +23,41 @@ def test_get_with_mapping():
         "capacity": 100.0,
     }
     mapped = {}
-    for key, value in expected.items():
+    for key, _ in expected.items():
         mapped[key] = mapper.get(key)
     assert mapped == expected
 
 
 def test_get_defaults():
```

Comment:
This tests includes too many setups and assumptions.
I would instead only test outputs of Mapper class with custom mapping, like:
```
mapping= {"DEFAULT": {...}, "Adapter": {...}}
mapper = Mapper(Adapter, mapping=mapping)
assert ...
```
This makes the test independent from changes in adapter class aka `parametrize_dataclass` or `as_dict`...

Reply:
Don't really understand your comment. Isn't that what's `test_get_with_mapping` is doing? I guess you talk about `test_get_defaults` which tests much more.

Reply:
yes, i meant `test_get_defaults` - don't know why I selected the wrong lines...
So my comment is, that `test_get_defaults` does too much (too much dependencies).
Unit testing is about testing small independent units

Reply:
removed it